INSP Apply assert arg on refactor of AssertEqualInspection

FatalCatharsis opened the pull request:
Addresses issue #2042

Changes the AssertEqualInspection refactor suggestion to append the rest of the argument list.

Originally, applying the refactor would pull apart the binary expression and replace it with two args, but any subsequent args on the assert were deleted.

(this is my first PR to this project so I wanted to start small)

matklad commented on src/main/kotlin/org/rust/ide/inspections/RsAssertEqualInspection.kt:
```diff
@@ -71,12 +71,25 @@ class RsAssertEqualInspection : RsLocalInspectionTool() {
+                argPair.second
+            )
+
+            if (assertArgument.formatMacroArgList.isNotEmpty()) {
```
I think we could remove this `if`, because the loop would be empty anyway.

FatalCatharsis replied:
Yup, redundant check. 
1f7dd3a

matklad commented on src/main/kotlin/org/rust/ide/inspections/RsAssertEqualInspection.kt:
```diff
@@ -71,12 +71,25 @@ class RsAssertEqualInspection : RsLocalInspectionTool() {
-            args.add(argPair.second)
+            val newAssert = factory.createExpression("$assertName!()") as RsCallExpr
+
+            val argPair = comparedAssertArgs(assertArgument) ?: return null
```
It would be slightly more idiomatic to write `let (left, right) = comparedAssert`

FatalCatharsis replied:
Agreed, that's much nicer. First time I've used Kotlin, so I'm still picking things up. 
1f7dd3a

FatalCatharsis commented:
Not sure if I did the CLA right. I went through the docusign portal and filled it out earlier. Do I need to paste it anywhere here or email it to someone?

matklad commented:
>Do I need to paste it anywhere here or email it to someone?

No, just add yourself to CONTRIBUTORS.txt in this PR :)

FatalCatharsis commented:
Hey, I'm

FatalCatharsis commented:
Added my github username to CONTRIBUTORS.txt 8f7667a and added some tests 829c226. Had trouble understanding the syntax for a bit, but I got the gist. Question, do you know why the tests required that <caret> block at the end? My test kept failing without it.

matklad commented:
bors r+

Thanks!

> Question, do you know why the tests required that <caret> block at the end? My test kept failing without it.

The `<caret>` specifies the position of the editor caret in the text document. It is required because what the tests do is that they invoke the light-bulb <kdb>Alt+Enter</kbd> keybinding, which looks for suggestions for the current context around the caret
